### DISCUSSION TITLE
Running FMU in OMEdit Error

### DISCUSSION THREAD

**Discussion by sbar123** [OpenModelica/OpenModelica#10071] [Q&A] [answered]
Hi,

I am trying to run an FMU that was imported in OMEdit. The FMU was created in another software (Heeds). When i am trying to run a simulation i get following error:

> Simulation process failed. Exited with code 0xffffffffffffffff.
> fmi2SetTime failed with status : Error

> simulation terminated by an assertion at initialization
> module = HS_passive__1, log level = ERROR: [logStatusError][FMU status:Error] fmi2SetTime called while not in continuous or event mode
> module = HS_passive__1, log level = ERROR: [logStatusError][FMU status:Error] fmi2Terminate cannot be called in this mode`

I using Version 1.19.2. The FMU is Model Exchange 2.0
The FMU is running in other software like dymola.
I know that the FMU import is still somewhat experimental, but is there something that  i can do?

Thank you!
Sebastian

**Comment by adeas31** [comment]
See #5345

**Comment by AnHeuermann** [comment, marked as answer]
You can try [SSP models](https://ssp-standard.org/) with [OMSimulator](https://openmodelica.org/doc/OMSimulator/master/html/) instead of the (old and should be deprecated) FMU import in OMEdit.

In OMEdit you can create a new SSP model with `File->New->New SSP Model.` Select `Strongly Coupled` for ModelExchange FMUs and `Weakly Coupled` for CoSimulation FMUs. Then you can add your FMU as submodule (the `Add SubModule` button) and after that hit the simulate button.

![grafik](https://user-images.githubusercontent.com/38031952/213428088-4804b572-8909-48ed-9846-5080da502bba.png)

If you want to connect a Modelica model `A` and your FMU `B` you can export `A` to an FMU first and than connect them in the SSP model.
There are some examples on how to work with SSP in https://github.com/OpenModelica/OMTeaching/tree/master/Tutorials/MODPROD22/Tutorial2.

**Reply by sbar123** [reply]
Thank you for the input! 
First things first: I managed to run the FMU in an SSP. 
That brings me to the next question:
I am currently running an hierachical model consisting out of several other models. Some of the  models are parametric. My plan was to replace some of the models by FMUs. As I inderstand i would have to export now all the models at the same level as the FMU to FMUs and connect them in the SSP-Model?

Another question:
Following the example from the MODPROD22 It tried to PIP OMSimulator with python (not Anaconda!) which failed with the following message:

`  × python setup.py bdist_wheel did not run successfully.
  │ exit code: 1
  ╰─> [8 lines of output]
      running bdist_wheel
      running build
      running build_py
      creating build
      creating build\lib
      creating build\lib\OMSimulator
      Starting new HTTPS connection (1): build.openmodelica.org:443
      error: HTTPSConnectionPool(host='build.openmodelica.org', port=443): Max retries exceeded with url: /omsimulator/nightly/win-msvc64/OMSimulator-win64-v2.1.1.post190-gcfb740c.zip (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x000001F8F4D5C040>: Failed to establish a new connection: [WinError 10060] Ein Verbindungsversuch ist fehlgeschlagen, da die Gegenstelle nach einer bestimmten Zeitspanne nicht richtig reagiert hat, oder die hergestellte Verbindung war fehlerhaft, da der verbundene Host nicht reagiert hat'))
      [end of output]`

This might be to our firewall. Although normally PIPing packages with the --proxy extension works. But it seems that it still tries to contact openmodellica.org during the install which might not be throught the proxy. Is there a way to offline install? will also try it on a different computer without the firewall.

Thanks again!
Regards,
Sebastian

**Reply by casella** [reply]
> Thank you for the input! First things first: I managed to run the FMU in an SSP. That brings me to the next question: I am currently running an hierachical model consisting out of several other models. Some of the models are parametric. My plan was to replace some of the models by FMUs. As I inderstand i would have to export now all the models at the same level as the FMU to FMUs and connect them in the SSP-Model?

With the current OMSimulator, yes. Maybe @lochel can comment on that.

> Another question: Following the example from the MODPROD22 It tried to PIP OMSimulator with python (not Anaconda!) which failed with the following message:
> 
> ` × python setup.py bdist_wheel did not run successfully. │ exit code: 1 ╰─> [8 lines of output] running bdist_wheel running build running build_py creating build creating build\lib creating build\lib\OMSimulator Starting new HTTPS connection (1): build.openmodelica.org:443 error: HTTPSConnectionPool(host='build.openmodelica.org', port=443): Max retries exceeded with url: /omsimulator/nightly/win-msvc64/OMSimulator-win64-v2.1.1.post190-gcfb740c.zip (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x000001F8F4D5C040>: Failed to establish a new connection: [WinError 10060] Ein Verbindungsversuch ist fehlgeschlagen, da die Gegenstelle nach einer bestimmten Zeitspanne nicht richtig reagiert hat, oder die hergestellte Verbindung war fehlerhaft, da der verbundene Host nicht reagiert hat')) [end of output]`
 
> This might be to our firewall.

That's likely, I'm not much into German but I understand there were some connection timeouts, right? (too lazy to pass it through Google Traslate)

> Although normally PIPing packages with the --proxy extension works. But it seems that it still tries to contact openmodellica.org during the install which might not be throught the proxy. Is there a way to offline install?

Whoa, probably not when using Python. We only implemented a backup solution to install the MSL with the package manager absent a viable network connection, but for everything else I guess you need it. If that is a problem, we can see how to work around it.

**Reply by lochel** [reply]
@sbar123 There is also some effort to integrate Modelica into SSP which might be useful for your use-case. However, that’s of course not available as of now.

**Comment by casella** [comment]
See also #9918. As @AnHeuermann noticed, if you want to run the FMU by itself, or connected to other FMUs, OMSimulator is the right tool. Importing FMUs in Modelica makes sense if you need an FMU inside a bigger Modelica model.

**Reply by casella** [reply]
However, there are currently some issues the FMU importing, that we still haven't figured out how to fix, see #9918. We're working on it.
